tests/test_publish.py

```diff
@@ -13,70 +17,88 @@ def simulated_nidigital_sessions(standalone_tsm_context: SemiconductorModuleCont
+        test_data = random.random()
+        pin_query_context.publish(test_data)
         published_data = published_data_reader.get_and_clear_published_data()
-        assert len(published_data) == num_sites
+        assert len(published_data) == self._NUM_SITES
         for published_data_point in published_data:
-            assert published_data_point.double_value == 1150.0
+            assert published_data_point.double_value == test_data
 
-    def test_publish_bool_scalar(self, pin_query_context, published_data_reader, num_sites):
+    def test_publish_bool_scalar(self, pin_query_context, published_data_reader):
         pin_query_context.publish(True)
         published_data = published_data_reader.get_and_clear_published_data()
-        assert len(published_data) == num_sites
+        assert len(published_data) == self._NUM_SITES
         for published_data_point in published_data:
             assert published_data_point.boolean_value
 
 
 @pytest.mark.pin_map("publish.pinmap")
 @pytest.mark.usefixtures("simulated_nidigital_sessions")
 class TestSinglePin1d:
+    _PIN = "DUTPin1"
+    _NUM_SITES = 3
+
     @pytest.fixture
     def pin_query_context(self, standalone_tsm_context):
-        pin_query_context, *_ = standalone_tsm_context.pins_to_nidigital_session_for_ppmu("DUTPin1")
+        pin_query_context, *_ = standalone_tsm_context.pins_to_nidigital_session_for_ppmu(self._PIN)
         return pin_query_context
 
     def test_publish_float_1d(self, pin_query_context, published_data_reader):
-        test_data = [1150.0, 1952.5]
+        test_data = [random.random() for _ in range(self._NUM_SITES)]
         pin_query_context.publish(test_data)
         published_data = published_data_reader.get_and_clear_published_data()
         assert len(published_data) == len(test_data)
         for published_data_point, test_data_point in zip(published_data, test_data):
             assert published_data_point.double_value == test_data_point
 
     def test_publish_bool_1d(self, pin_query_context, published_data_reader):
-        test_data = [True, False]
+        test_data = [bool(i % 2) for i in range(self._NUM_SITES)]
         pin_query_context.publish(test_data)
         published_data = published_data_reader.get_and_clear_published_data()
         assert len(published_data) == len(test_data)
         for published_data_point, test_data_point in zip(published_data, test_data):
             assert published_data_point.boolean_value == test_data_point
 
+    def test_publish_pattern(self, standalone_tsm_context, published_data_reader):
+        (
+            pin_query_context,
+            session,
+            site_list,
+        ) = standalone_tsm_context.pins_to_nidigital_session_for_pattern(self._PIN)
+        num_sites = len(site_list.split(","))
+        # alternate True and False
+        test_data = collections.OrderedDict((i, bool(i % 2)) for i in range(num_sites))
```

Comment:
I'd like to create the dictionary with a different ordering of sites than what TSM uses just to verify that our data transformation algorithm is correct. 
After reviewing the other unit tests I decided that I prefer that we use a simple definition of test data instead of generating it with code. For example: `test_data = {1:False, 0:True, 2:True}`
To validate the data, use the `site_numbers` property:
```
for published_data_point, site_number in zip(published_data, site_numbers)
    assert published_data_point.boolean_value == test_data[site_number]
```

Reply:
20c4a952b4a837227c4df69a06962da870e5d30b